metrics/docs/01_storing_database_metadata.md

```diff
@@ -0,0 +1,115 @@
+
+Let's review the way of storing the database metadata: 
+
+## Database Metadata
+> Explain the way of storing the database metadata in the database.
+
+### Firestore 
+> Explain the database structure and rules needed to store the metadata in the Firestore.
+
+To store the database metadata and make it easily accessible, we need to add to the `metadata` collection with the `metadata` document. This document will contain the `databaseVersion` field of `String` and `isUpdating` `boolean` field. 
+
+So, the database structure will look like the following: 
+
+> - projects
+> - ...
+> - metadata
+>   - metadata
+>       - databaseVersion: String
+>       - isUpdating : bool
+
+
+The `metadata` collection should have the following security rules: 
+
+- No one can write (update, delete, create) documents in this collection.
+- Anybody can read documents from this collection. 
+
+## Application Version
+> Explain the way of storing the application version and providing it to the Metrics applications.
+
+To detect whether the current application version is compatible with the database, we should have the current application version. Since this value is common for database and Metrics applications, we should make it easily accessible in any application like `CI Integrations`, `Metrics CLI`, or `Metrics Web`. 
+
+To do so, we can store the application version in the `version` file under the [metrics](https://github.com/platform-platform/monorepo/tree/master/metrics) package of our repository. It allows us to get the contents of this file and pass it as an environment variable to any of our applications during the building process.
+
+### Metrics Web Application
+> Explain the way of passing the application version to the Metrics Web Application.
+
+Let's consider an example of a Unix shell script for getting the application version from the file and passing it to the `Metrics Web Application` as an environment variable.
+
+Assume we run the following script under the `metrics/web` package: 
+
+```bash
+VERSION=$(cat ../version)
+
+flutter build web --release --dart-define=APP_VERSION=$VERSION
+```
+
+So, we can use the following line of code to get the application version in the Metrics Web Application: 
+
+```dart
+const appVersion = String.fromEnvironment('APP_VERSION');
+```
+
+Also, the `version` file will simplify the process of getting the latest version during database deployment/updating to set the current database version.
+
+## Making things work
+> Explain the process of updating the Metrics applications.
+
+Once we have a database version in the Firestore database and an application version, we should block the Metrics applications from updating the database records when the database is updating or its version is not compatible with the current application version.
+
+Let's consider the process of blocking the Metrics Web Application: 
+
+### Metrics Web Application
```

Comment:
Should we add some notes about CLI like `ci_integrations` how it should use version, etc...?